src/operation.ts

```diff
@@ -71,8 +72,28 @@ export function sendTransaction<T>(
       tx = transaction
     }
 
+    const from = await context.getAccount().pipe(first()).toPromise()
+    let gasEstimate: number = 0
+    try {
+      gasEstimate = await tx.estimateGas()
+    } catch (error) {
+      let errToReturn: Error
+      try {
+        errToReturn = await errorHandler(error)
+      } catch (err) {
+        errToReturn = err
+      }
+      observer.error(errToReturn)
+    }
+    let gas: number
+    if (gasEstimate) {
+      gas = new BN(gasEstimate * 1.1)
+    } else {
+      gas =  new BN(6000000)
```

Comment:
Why use such a big number?
We don't have any action in that system that requires such an amount of gas.

Reply:
It's just a "big enough" number. Do we know what the maximum amount of gas is for actions in our system so we can set it lower?

Reply:
I'm sure 1M is big enough.